src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCache.cs

```diff
@@ -79,12 +79,7 @@ public SimpleQueueCache(int cacheSize, Logger logger)
         /// </summary>
         public virtual bool IsUnderPressure()
         {
-            if (cachedMessages.Count == 0) return false; // empty cache
-            if (Size < maxCacheSize) return false; // there is still space in cache
-            if (cacheCursorHistogram.Count == 0) return false;    // no cursors yet - zero consumers basically yet.
-            // cache is full. Check how many cursors we have in the oldest bucket.
-            int numCursorsInLastBucket = cacheCursorHistogram[0].NumCurrentCursors;
-            return numCursorsInLastBucket > 0;
+            return cacheCursorHistogram.Count >= NUM_CACHE_HISTOGRAM_BUCKETS;
```

Comment:
This change addresses the bug.
It addresses it by modifying IsUnderPressure to report correctly given the GetMaxAddCount.

The back-off logic relies on both of these calls, and requires them to function in tandem.  Which call was behaving poorly is a matter of interpretation.  I've opted to address the problem by simplifying the more confusing of the two.

You're suggestion is to correct the bug by making GetMaxAddCount more complicated (admittedly trivially so) rather than simplifying the IsUnderPressure.